File: android/src/main/java/com/video/hybrids/HybridVideoViewViewManagerFactory.kt
package com.margelo.nitro.video

import com.facebook.proguard.annotations.DoNotStrip

@DoNotStrip
class HybridVideoViewViewManagerFactory: HybridVideoViewViewManagerFactorySpec() {
  override fun createViewManager(nitroId: Double): HybridVideoViewViewManagerSpec {
    return HybridVideoViewViewManager(nitroId.toInt())
  }

  override val memorySize: Long
    get() = 0
}

File: ios/hybrids/HybridVideoViewViewManagerFactory.swift
//
//  HybridVideoViewViewManagerFactory.swift
//  NitroVideo
//
//  Created by Krzysztof Moch on 23/09/2024.
//

import Foundation

class HybridVideoViewViewManagerFactory: HybridVideoViewViewManagerFactorySpec {
  func createViewManager(nitroId: Double) throws -> any HybridVideoViewViewManagerSpec {
    return try HybridVideoViewViewManager(nitroId: nitroId)
  }
  
  // Initialize HybridContext
  var hybridContext = margelo.nitro.HybridContext()
  
  // Return size of the instance to inform JS GC about memory pressure
  var memorySize: Int {
    return getSizeOf(self)
  }
}
